src/Illuminate/Database/Query/Builder.php

```diff
@@ -3677,8 +3683,8 @@ protected function setAggregate($function, $columns)
      *
      * After running the callback, the columns are reset to the original value.
      *
-     * @param  array  $columns
-     * @param  callable  $callback
+     * @param  array<ColumnTypes>  $columns
+     * @param  callable(): mixed  $callback
      * @return mixed
```

Comment:
You could probably make this generic:
```suggestion
     * @template T
     * @param  array<ColumnTypes>  $columns
     * @param  callable(): T  $callback
     * @return T
```